### PR TITLE
VTOL Tailsitter: add automatic pitch ramps also in Stabilized mode

### PR DESCRIPTION
**Pull request by sfuhrer**
### Solved Problem
Fixes https://github.com/PX4/PX4-Autopilot/issues/22316

Tailsitter transitions in Stabilized are currently handed the following way:
- transition to FW: user has to manually pitch down until within 5° of the max tilt in hover, MPC_MAN_TILT_MAX
- transition to MC: the pitch setpoint immediately changes to the hover interpretation, meaning that if the user doesn't pitch down while doing the transition then the vehicle will immediately try to pitch up to 0° hover pitch (so a 90° step)

### Solution
Do the same pitch ramp as in all altitude controlled modes also in Stabilized. 

### Changelog Entry
For release notes:
```
Feature: VTOL Tailsitter: add automatic pitch ramps also in Stabilized mode
Documentation: Need to clarify page ... / done, read docs.px4.io/...
```

### Test coverage
Flight tested. 

### Context
Screen recording of a SITL flight with the changes as proposed in this PR, and at the end a failed transition with current main (there also the SITL tuning is bad, also fixed it in this PR).
https://github.com/PX4/PX4-Autopilot/assets/26798987/2eff901a-2aa1-484a-a0e5-116aad1c61c5

### CODE REVIEW

File: src/modules/mc_att_control/mc_att_control_main.cpp [hunk 1]
```diff
@@ -193,9 +193,11 @@ MulticopterAttitudeControl::generate_attitude_setpoint(const Quatf &q, float dt,
 	attitude_setpoint.timestamp = hrt_absolute_time();
 	_vehicle_attitude_setpoint_pub.publish(attitude_setpoint);
 
-	// update attitude controller setpoint immediately
-	_attitude_control.setAttitudeSetpoint(q_sp, attitude_setpoint.yaw_sp_move_rate);
```
Comment (sfuhrer) [inline src/modules/mc_att_control/mc_att_control_main.cpp:197]: @MaEtUgR I guess it's done on purpose that in the MC attitude controller, when flying in Stabilized, the attitude setpoint that is generated here is directly sent to the attitude controller, and not passed over the `attitude_setpoint` uORB topic?

File: src/modules/vtol_att_control/tailsitter.cpp [hunk 2]
```diff
@@ -206,10 +198,8 @@ void Tailsitter::update_transition_state()
 	_q_trans_sp.normalize();
 
 	// tilt angle (zero if vehicle nose points up (hover))
-	float cos_tilt = _q_trans_sp(0) * _q_trans_sp(0) - _q_trans_sp(1) * _q_trans_sp(1) - _q_trans_sp(2) *
-			 _q_trans_sp(2) + _q_trans_sp(3) * _q_trans_sp(3);
-	cos_tilt = cos_tilt >  1.0f ?  1.0f : cos_tilt;
-	cos_tilt = cos_tilt < -1.0f ? -1.0f : cos_tilt;
+	const float cos_tilt = math::constrain(_q_trans_sp(0) * _q_trans_sp(0) - _q_trans_sp(1) * _q_trans_sp(1) -
```
Comment (RomanBapst) [inline src/modules/vtol_att_control/tailsitter.cpp:201]: @sfuhrer Could be put into the math library. Getting the tilt from a quaterion seems to be something useful.

File: src/modules/vtol_att_control/tailsitter.cpp [hunk 3]
```diff
@@ -172,31 +165,30 @@ void Tailsitter::update_transition_state()
...
 			_q_trans_start = _q_trans_start * Quatf(Eulerf(0, -M_PI_2_F, 0));
 
 		} else if (_vtol_mode == vtol_mode::TRANSITION_FRONT_P1) {
 			// initial attitude setpoint for the transition should be with wings level
-			_q_trans_start = Eulerf(0.0f, _mc_virtual_att_sp->pitch_body, _mc_virtual_att_sp->yaw_body);
-			Vector3f x = Dcmf(Quatf(_v_att->q)) * Vector3f(1, 0, 0);
-			_trans_rot_axis = -x.cross(Vector3f(0, 0, -1));
+			_q_trans_start = Eulerf(0.f, _mc_virtual_att_sp->pitch_body, _mc_virtual_att_sp->yaw_body);
+			Vector3f x = Dcmf(Quatf(_v_att->q)) * Vector3f(1.f, 0.f, 0.f);
+			_trans_rot_axis = -x.cross(Vector3f(0.f, 0.f, -1.f));
```
Comment (RomanBapst) [inline src/modules/vtol_att_control/tailsitter.cpp:191]: @sfuhrer Can be for another day bould I would love to add some convenience functions to the mathlib where you can specify the current attitude and the target attitude and you get the intermediate quaternions along the shorted rotation.
Then we can remove all of this untested and hard to understand math.

**Review by RomanBapst — APPROVED** [review]
@sfuhrer Changes look good, I think we should at some point move as much of the quaternion math into a lib. Not a blocker for this PR.

**Comment by sfuhrer** [comment]
> @sfuhrer Changes look good, I think we should at some point move as much of the quaternion math into a lib. Not a blocker for this PR.

Yes very good points, that should be done next.


I fixed a format failure and force pushed.

**Review by RomanBapst — APPROVED** [review]

**Pull request merged**